spirv_reflect.h

```diff
@@ -340,13 +341,19 @@ typedef struct SpvReflectTypeDescription {
     SpvReflectArrayTraits           array;
   } traits;
 
+  // If underlying type is a struct (ex. array of structs)
+  // this gives access to the OpTypeStruct
+  struct SpvReflectTypeDescription* struct_type_description;
+
+  // @deprecated use struct_type_description instead
   uint32_t                          member_count;
+  // @deprecated use struct_type_description instead
```

Comment:
How does struct_type_description account for the struct members? I don't see any code that parses the members vars.

Reply:
So we call `ParseType` on all `OpTypeStructs` and build a `SpvReflectTypeDescription` for every struct already (and that is what is parsing the member vars)

Because SSA enforce ordering, when we get to the `ParseType` for a `OpTypeArray`, just calling `FindType()` will always find the struct the array is made up of

-----

printing from GDB

- `p reflection.m_module.input_variables[0].name` == `"int_b"`
- `p reflection.m_module.input_variables[0].type_description->type_name` == `"structX"`

so from the `SpvReflectTypeDescription` object

- `p type_description->struct_type_description.type_name` == `"structX"`
- `p type_description->struct_type_description.members[1].struct_member_name` = `"x2_"`
- `p type_description->members[1].struct_member_name` = `"x2_"`